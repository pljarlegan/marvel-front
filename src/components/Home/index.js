import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';

const useStyles = makeStyles({
  card: {
    maxWidth: 345,
  },
});

export default function Home() {
  const classes = useStyles();

  return (
    <Grid
      container
      spacing="0"
      direction="column"
      alignItems="center"
      justify="center"
    >
      <Grid item xs="12">
        <Card>
          <Card className={classes.card}>
            <CardActionArea>
              <CardMedia
                component="img"
                alt="Contemplative Reptile"
                height="140"
                image="/marvel.jpg"
                title="Contemplative Reptile"
              />
              <CardContent>
                <Typography gutterBottom variant="h5" component="h2">
                  Marvel
                </Typography>
                <Typography variant="body2" color="textSecondary" component="p">
                  <b>Marvel Studios</b> est une société de production cinématographique américaine en prises de vue réelles et en animation assurant les adaptations cinématographiques des productions de Marvel Entertainment, essentiellement issues des Marvel Comics.
                </Typography>
              </CardContent>
            </CardActionArea>
          </Card>
        </Card>
      </Grid>
    </Grid>

  );
}
