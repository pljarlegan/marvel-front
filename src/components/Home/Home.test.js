import React from 'react';
import ReactDOM from 'react-dom';
import Home from './index';
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea"
import { shallow } from "enzyme";
import { configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Home/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should render 2 cards', () => {
    const wrapper = shallow(<Home/>);
    expect(wrapper.find(Card)).toHaveLength(2);
  });

  it('should render a CardActionArea', () => {
    const wrapper = shallow(<Home/>);
    expect(wrapper.find(CardActionArea)).toHaveLength(1);
  });

  it("Matches the snapshot", () => {
    const wrapper = shallow(<Home/>);
    expect(wrapper).toMatchSnapshot();
  });
});
