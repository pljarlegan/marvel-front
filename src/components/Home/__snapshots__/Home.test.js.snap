// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Matches the snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  direction="column"
  justify="center"
  spacing={0}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Card))>
      <WithStyles(ForwardRef(Card))
        className="makeStyles-card-173"
      >
        <WithStyles(ForwardRef(CardActionArea))>
          <WithStyles(ForwardRef(CardMedia))
            alt="Marvel"
            component="img"
            height="140"
            image="/MAD.jpg"
            title="Marvel"
          />
          <WithStyles(ForwardRef(CardContent))>
            <WithStyles(ForwardRef(Typography))
              component="h2"
              gutterBottom={true}
              variant="h5"
            >
              MAD : Marvel Api Demo
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Typography))
              color="textSecondary"
              component="p"
              variant="body2"
            >
              <b>
                Marvel Studios
              </b>
               est une société de production cinématographique américaine en prises de vue réelles et en animation assurant les adaptations cinématographiques des productions de Marvel Entertainment, essentiellement issues des Marvel Comics.
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(CardContent))>
        </WithStyles(ForwardRef(CardActionArea))>
      </WithStyles(ForwardRef(Card))>
    </WithStyles(ForwardRef(Card))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
